instrumentation/jdbc/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/jdbc/JdbcSingletons.java

```diff
@@ -68,5 +71,31 @@ public static Instrumenter<DataSource, DbInfo> dataSourceInstrumenter() {
     return DATASOURCE_INSTRUMENTER;
   }
 
+  private static final Cache<Class<?>, Boolean> wrapperClassCache = Cache.weak();
```

Comment:
I don't have great intuition about how many wrappers might be in an application, but should we make this bounded?

```suggestion
  private static final Cache<Class<?>, Boolean> wrapperClassCache = Cache.bounded(100);
```

Reply:
The key is a class, the number of classes isn't going to be too high.